src/message.c

```diff
@@ -1301,6 +1301,10 @@ qd_message_t *qd_message_receive(pn_delivery_t *delivery)
         msg->strip_annotations_in  = qd_connection_strip_annotations_in(qdc);
         pn_record_def(record, PN_DELIVERY_CTX, PN_WEAKREF);
         pn_record_set(record, PN_DELIVERY_CTX, (void*) msg);
+        msg->content->max_message_size = qd_connection_max_message_size(qdc);
+    } else {
+        // existing messages may detect oversize transitions only once
+        msg->content->oversize_detected = false;
```

Comment:
Is setting msg->content->oversize_detected  to false necessary ? 
Calling ZERO (msg); already sets this flag to false. Also, once this flag becomes true it *should not* go to false?

Reply:
After further consideration here's another strategy:
Instead of two flags use a single counter: Upon going oversize the count will go from zero to one. When the count is zero then the message is not oversize. When the count is one that's when router_node.c takes "just went oversize" action. When the count is greater than one then router_node.c takes no special action.

This would probably be a performance improvement as the counter is modified only when oversize messages are being handled.